src/App.tsx

```diff
@@ -15,47 +15,73 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABETICALLY,
+  LENGTH,
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const [isReversed, setIsReversed] = useState(false);
+  const sortGoods = (a: string, b:string):number => {
+    if (sortType === SortType.NONE) {
+      return 0;
+    }
+
+    return sortType === SortType.ALPHABETICALLY
+      ? a.localeCompare(b)
+      : a.length - b.length;
+  };
```

Comment:
your sortGoods should be in `useCallback` hook because it depends on sortType
or you can use the function `getReorderedGoods` from your previous task and just pass sortType as an argument

Reply:
moved up component function :)